File: WebhookTag/src/main/java/com/aliucord/plugins/WebhookTag.java
package com.aliucord.plugins;

import android.annotation.SuppressLint;
import android.content.Context;
import android.widget.TextView;

import androidx.annotation.NonNull;

import com.aliucord.Logger;
import com.aliucord.entities.Plugin;
import com.aliucord.patcher.PinePatchFn;
import com.discord.api.user.User;
import com.discord.models.message.Message;
import com.discord.models.user.CoreUser;
import com.discord.widgets.chat.list.adapter.WidgetChatListAdapterItemMessage;

@SuppressWarnings("unused")
public class WebhookTag extends Plugin {
  public Logger logger = new Logger("WebhookTags");

  @NonNull
  @Override
  public Manifest getManifest() {
    var manifest = new Manifest();
    manifest.authors = new Manifest.Author[] {new Manifest.Author("Nat Sepruko", 156990761366192128L) };
    manifest.description = "Sets the text of the Bot tag on webhooks to \"WEBHOOK\"";
    manifest.version = "1.0.1";
    manifest.updateUrl = "https://raw.githubusercontent.com/NatSepruko/aliucord-plugins/builds/updater.json";
    return manifest;
  }

  @SuppressLint("SetTextI18n")
  @Override
  public void start(Context context) throws Throwable {
    var tag = WidgetChatListAdapterItemMessage.class.getDeclaredField("itemTag");
    tag.setAccessible(true);

    patcher.patch(WidgetChatListAdapterItemMessage.class, "configureItemTag", new Class<?>[] { Message.class }, new PinePatchFn(callFrame -> {
      Message msg = (Message) callFrame.args[0];
      User author = msg.getAuthor();
      CoreUser coreUser = new CoreUser(author);

      try {
        TextView textView = (TextView) tag.get(callFrame.thisObject);
        if (msg.getWebhookId() != null && coreUser.getDiscriminator() == 0 && textView != null) textView.setText("WEBHOOK");
      } catch(Throwable e) {
        logger.error("Error setting tag to WEBHOOK", e);
      }
    }));
  }

  @Override
  public void stop(Context context) {
    patcher.unpatchAll();
  }
}

File: WebhookTag/src/main/java/com/aliucord/plugins/WebhookTag.kt
package com.aliucord.plugins

import android.annotation.SuppressLint
import android.content.Context
import android.widget.TextView
import com.aliucord.Logger
import com.aliucord.entities.Plugin
import com.aliucord.entities.Plugin.Manifest.Author
import com.aliucord.patcher.PinePatchFn
import com.discord.api.user.User
import com.discord.models.message.Message
import com.discord.widgets.chat.list.adapter.WidgetChatListAdapterItemMessage
import top.canyie.pine.Pine.CallFrame
import com.discord.models.user.CoreUser


class WebhookTag : Plugin() {
    var logger = Logger("WebhookTags")

    override fun getManifest() =
        Manifest().apply {
            authors = arrayOf(Author("Nat", 156990761366192128L))
            description = "Replaces the \"BOT\" text with \"WEBHOOK\" on webhooks."
            version = "1.1.1"
            updateUrl = "https://raw.githubusercontent.com/NatSepruko/aliucord-plugins/builds/updater.json"
            changelog =
                """
                    Improved {improved marginTop}
                    ======================
                    
                    * Migrated to Kotlin because it's better
                    * Updated the plugin description
                    
                    Fixed {fixed marginTop}
                    ======================
                    
                    * Re-add null-check because I didn't quite know how null-checking worked with Kotlin in this situation
                """.trimIndent()
        }

    @SuppressLint("SetTextI18n")
    override fun start(context: Context?) {
        patcher.patch(
            WidgetChatListAdapterItemMessage::class.java,
            "configureItemTag",
            arrayOf(Message::class.java),
            PinePatchFn{ callFrame: CallFrame ->
                val msg = callFrame.args[0] as Message
                val author = msg.author as User
                val coreUser = CoreUser(author)

                val tag = callFrame.thisObject.javaClass
                    .getDeclaredField("itemTag")
                    .let {
                        it.isAccessible = true
                        it.get(callFrame.thisObject) as TextView?
                    }

                try {
                    if (msg.webhookId != null && coreUser.discriminator == 0 && tag != null) tag.text = "WEBHOOK"
                } catch (e: Throwable) {
                    logger.error("Failed to set text on webhook tag", e)
                }
            })
    }

    override fun stop(context: Context?) {
        patcher.unpatchAll()
    }

}